main.go

```diff
@@ -82,6 +88,15 @@ func init() {
 	flag.Set("alsologtostderr", "true") // for klog
 	flag.Parse()
 
+	commitHash, commitTime, dirtyBuild := getBuildInfo()
+	arch := fmt.Sprintf("%v/%v", runtime.GOOS, runtime.GOARCH)
+
+	fmt.Printf("gen-crd-api-reference-docs version=%s commit=%s date=%s dirty=%v arch=%s go=%v\n", version, commitHash, commitTime, dirtyBuild, arch, runtime.Version())
```

Comment:
Should we move this inside the if?

Reply:
I was thinking that it would be useful to always print the version at startup, but happy to move it.

Reply:
I forgot how much output the program prints. If it's very little, it might get annoying.

Reply:
it would be something like

```
❯ gen-crd-api-reference-docs -config site/config/api/config.json -template-dir site/config/api -out-file site/content/reference/api.md -api-dir "github.com/nginxinc/nginx-gateway-fabric/apis"
gen-crd-api-reference-docs version=0.3.0-SNAPSHOT-62fa6e2 commit=62fa6e27a07bac0cf553ec35ab52736d3a58a5e6 date=2024-07-24T22:16:31Z dirty=false arch=darwin/arm64 go=go1.22.5
I0724 19:01:55.984928   60449 main.go:144] parsing go packages in directory github.com/nginxinc/nginx-gateway-fabric/apis
I0724 19:02:00.058290   60449 main.go:246] using package=github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1
I0724 19:02:00.070753   60449 main.go:182] written to site/content/reference/api.md
```

Reply:
Ehh looks a bit wonky especially since we use klog for the rest of the output. Let's move inside "if" IMO if you don't mind. 🤞

Reply:
done